python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -299,20 +300,31 @@ def calculate_used_data(
                     if upstream_key in unknown_required_keys:
                         upstream_required_keys.add(upstream_key)
 
-                # get the most recent asset materialization for this parent which happened before
-                # the current record
-                latest_parent_record = self.get_latest_materialization_record(
-                    parent_key, before_cursor=record_id
-                )
+                input_event_pointer_tag = get_input_event_pointer_tag_key(parent_key)
+                if input_event_pointer_tag in record_tags:
+                    # get the asset materialization record that was calculated using the logical
```

Comment:
nitpick: even though we added this tag in the same PRs that we added logical versioning, it stands on its own independent of logical versioning. I think this comment makes what's going on here sound fancier / more confusing than what actually is.